### PR TITLE
tsdb: add pool for RefSeries to reduce memory allocation

### PR DESCRIPTION
**Pull request by oabunk**
@@bwplotka @codesome PTAL.

### CODE REVIEW

File: tsdb/head.go [hunk 1]
```diff
@@ -1029,8 +1030,9 @@ func (h *Head) appender() *headAppender {
 		minValidTime:          max(h.minValidTime.Load(), h.MaxTime()-h.chunkRange.Load()/2),
 		mint:                  math.MaxInt64,
 		maxt:                  math.MinInt64,
+		series:                h.getSeriesBuffer(),
```
Comment (bwplotka) [inline tsdb/head.go:1033]:
```suggestion
		series:                h.getSeriesRefsBuffer(),
```

**Review by bwplotka — COMMENTED** [review]
Thanks, generally it looks ok, but can we have some benchmarks/run existing micro-benchmarks to validate that this actually helps? All performance related changes should be backed up by results so we can make data-driven decision (: WDYT?

**Comment by codesome** [comment]
@oabunk if you have done any benchmarks, could you please share the results. Also, for this to show any different, there needs to be deletion of series along with creation.

I have to check the implementation still, but something does not feel right about putting the memSeries back to the pool on commit.

**Comment by bwplotka** [comment]
Any update? :hugs:

**Comment by roidelapluie** [comment]
We have looked at this pull request during our bug scrub.

We have decided to close this pull request due to the lack of benchmarks.

Thank you for your contribution.

**Pull request closed without merging**